scripts/validate-artifacts.ps1

```diff
@@ -1,121 +1,75 @@
-        if ($storageEstimate.'Region' -ne 'UK West') { 
-            throw "Unable to verify the VM region - please make sure it set to UK West and try again."
-        } 
-
-        if (-not $storageEstimate.Description.Contains("Managed Disks, Premium SSD, LRS Redundancy")) { 
-            throw "Unable to verify disk type - please make sure that you added a mannaded disk with proper disk type (Premium SSD) and replication type (LRS) to the calculations and try again."
-        } 
-        
-        if (-not $storageEstimate.Description.Contains("P6")) { 
-            throw "Unable to verify data disk size - please make sure that disk size for data disk is set to 64GB and try again. "
-        } 
+# generation of artifacts
+Write-Output "Generating artifacts"
 
-        Write-Output "`u{2705} Checked calculations for the data disk (mannaged disk) - OK."
+Write-Output "Checking if temp folder exists"
+if (-not (Test-Path "$tempFolderPath")) { 
+    Write-Output "Temp folder does not exist, creating..."
+    New-Item -ItemType Directory -Path $tempFolderPath
+}
 
-    } else { 
-        throw "Unable to verify estimates for the category 'Storage' - more than one item found. Please make sure that you have only one separate storage resource (for your data disk) in your calculations and try again."
-    }
+Write-Output "Exporting resources template"
+Export-AzResourceGroup -ResourceGroupName $rgName -Path "$tempFolderPath/$resourcesTemplateName" -Force
 
-} else { 
-    throw "Unable to find estimates for the category 'Storage. Please make sure that you added your data disk to the pricing calculation."
+Write-Output "Uploading resources template"
+$ResourcesTemplateBlob = @{
+    File             = "$tempFolderPath/$resourcesTemplateName"
+    Container        = $containerName
+    Blob             = "$taskName/$resourcesTemplateName"
+    Context          = $storageAccount.Context
+    StandardBlobTier = 'Hot'
 }
+$blob = Set-AzStorageBlobContent @ResourcesTemplateBlob -Force
 
-$networkingEstimate = $data | Where-Object {$_.'Service Category' -eq 'Networking'}
-if ($networkingEstimate) { 
-    if ($networkingEstimate.'Service Category'.Count -eq 1) { 
-        
-        if ($networkingEstimate.'Service type' -ne 'IP Addresses') { 
-            throw "Unable to verify Public IP resource - please make sure that you added it to the calculations and try again."
-        }
-        
-        if ($networkingEstimate.'Region' -ne 'UK West') { 
-            throw "Unable to verify the Public IP region - please make sure it set to UK West and try again."
-        } 
-
-        Write-Output "`u{2705} Checked calculations for the Public IP address - OK."
+Write-Output "Generating a SAS token for the template artifact"
+$date = Get-Date
+$date = $date.AddDays(30) 
+$resourcesTemplateSaSToken = New-AzStorageBlobSASToken -Container $containerName -Blob "$taskName/$resourcesTemplateName" -Permission r -ExpiryTime $date -Context $storageAccount.Context
```

Comment:
The SAS token is generated with a 30-day expiry. Ensure this duration aligns with your security and access policies.